docs/concepts/validation_decorator.md

```diff
@@ -39,151 +39,83 @@ except ValidationError as exc:
+As with the rest of Pydantic, types are by default coerced by the decorator before they're passed to the actual function:
 
-```py test="no-print-intercept"
-# TODO replace find_file with something that isn't affected the filesystem
-import os
-from pathlib import Path
-from typing import Optional, Pattern
+```py
+from datetime import date
 
-from pydantic import DirectoryPath, validate_call
+from pydantic import validate_call
 
 
 @validate_call
-def find_file(path: DirectoryPath, regex: Pattern, max=None) -> Optional[Path]:
-    for i, f in enumerate(path.glob('**/*')):
-        if max and i > max:
-            return
-        if f.is_file() and regex.fullmatch(str(f.relative_to(path))):
-            return f
-
+def greater_than(d1: date, d2: date, *, include_equal=False) -> date:  # (1)!
+    if include_equal:
+        return d1 >= d2
+    else:
+        return d1 > d2
 
-# note: this_dir is a string here
-this_dir = os.path.dirname(__file__)
 
-print(find_file(this_dir, '^validation.*'))
-print(find_file(this_dir, '^foobar.*', max=3))
+d1 = '2000-01-01'  # (2)!
+d2 = date(2001, 1, 1)
+greater_than(d1, d2, include_equal=True)
 ```
 
-A few notes:
+1. Because `include_equal` has no type annotation, it will be inferred as [`Any`][typing.Any].
+2. Although `d1` is a string, it will be converted as a [`date`][datetime.date] object.
 
-- Though they're passed as strings, `path` and `regex` are converted to a `Path` object and regex respectively
-    by the decorator.
-- `max` has no type annotation, so will be considered as `Any` by the decorator.
+Type coercion like this can be extremely helpful, but also confusing or not desired. [Strict mode](strict_mode.md)
+can be enabled by using a [custom configuration](#custom-configuration).
 
-Type coercion like this can be extremely helpful, but also confusing or not desired.
-See [Coercion and strictness](#coercion-and-strictness) for a discussion of `@validate_call`'s limitations
-in this regard.
+!!! note "Validating the return value"
+    By default, the return value of the function is **not** validated. To do so, the `validate_return` argument
+    of the decorator can be set to `True`.
 
 ## Function signatures
 
-The `@validate_call` decorator is designed to work with functions using all possible parameter configurations and
-all possible combinations of these:
-
-* Positional or keyword arguments with or without defaults.
-* Variable positional arguments defined via `*` (often `*args`).
-* Variable keyword arguments defined via `**` (often `**kwargs`).
-* Keyword-only arguments: arguments after `*,`.
-* Positional-only arguments: arguments before `, /` (new in Python 3.8).
+The [`validate_call()`][pydantic.validate_call] decorator is designed to work with functions
+using all possible [parameter configurations][parameter] and all possible combinations of these:
 
-To demonstrate all the above parameter types:
```

Comment:
I'm inclined to remove this long example which imo doesn't really add any value, as such examples can be found in the linked Python docs above (https://docs.python.org/3/glossary.html#term-parameter).

Reply:
Honestly, let's leave it. I think it's sometimes helpful to do a bit of Python review to help folks who may be unfamiliar with such syntax, but realize it's relevant for their use case. I think it also makes people happy to see that we support such a wide variety of cases here.

That being said, I do think it'd be great if we could link to the relevant Python docs.

Reply:
In the API docs I would agree, but I think more in depth examples on the concepts page can be helpful 👍